encrypted-config-value/src/main/java/com/palantir/config/crypto/EncryptedValue.java

```diff
@@ -18,21 +18,32 @@
 
 import static com.google.common.base.Preconditions.checkArgument;
 
+import com.google.common.io.BaseEncoding;
 import com.palantir.config.crypto.algorithm.Algorithm;
 import com.palantir.config.crypto.algorithm.Algorithms;
 import java.io.IOException;
+import javax.annotation.Nullable;
 import org.immutables.value.Value;
 
 @Value.Immutable
 public abstract class EncryptedValue {
 
     private static final String PREFIX = "enc:";
 
-    public abstract String encryptedValue();
+    @Nullable
+    public abstract String getAlgorithm();
```

Comment:
Didn't see the second comment - we could fill in with `UNKNOWN` or something like that? We'd probably have to check for that in the `toString` impl so that old-style values round-trip through `EncryptedValue` without gaining a prefix.